// @wkoza/ngx-upload/esm2015/ngx-upload.module.js
import { InjectionToken, isDevMode, NgModule, Optional, SkipSelf } from '@angular/core';
import { NgxDragAndDropDirective } from './directives/dropzone.directive';
import { NgxThumbnailDirective } from './directives/thumbnail.directive';
import { NgxInputFileDirective } from './directives/inputfile.directive';
import { NGX_DROP_TARGET_OPTIONS, NGX_LOGGER_OPTIONS, ngxDropTargetOptions, ngxloggerOptions } from './utils/configuration.model';
import { ConsoleLogger, NgxUploadLogger, NoOpLogger } from './utils/logger.model';
import { InputfileComponent } from './components/inputfile.component';
const ngxDeclarations = [
    NgxDragAndDropDirective, NgxThumbnailDirective, InputfileComponent, NgxInputFileDirective
];
/**
 * Factory associated with internal logger
 * @param options
 * @returns {any}
 * @private
 */
export function _loggerFactory(options) {
    const enabled = options.enabled ? options.enabled : isDevMode();
    if (enabled) {
        const _console = typeof console === 'object' ? console : {};
        const debug = options.debug ? options.debug : true;
        return new ConsoleLogger(_console, debug);
    }
    return new NoOpLogger();
}
export const NGX_UPLOAD_ROOT_GUARD = new InjectionToken('Internal forRoot Guard');
export function createNgxUploadRootGuard(options) {
    if (options) {
        throw new TypeError('NgxUploadModule.forRoot() is called twice.');
    }
    return 'guarded';
}
export class NgxUploadModule {
    static forRoot(dropTargetOptions, loggerOptions) {
        return {
            ngModule: NgxUploadModule,
            providers: [
                { provide: NGX_LOGGER_OPTIONS, useValue: (loggerOptions) ? loggerOptions : ngxloggerOptions },
                {
                    provide: NGX_DROP_TARGET_OPTIONS,
                    useValue: (dropTargetOptions) ? dropTargetOptions : ngxDropTargetOptions
                },
                {
                    provide: NgxUploadLogger,
                    useFactory: _loggerFactory,
                    deps: [NGX_LOGGER_OPTIONS]
                },
                {
                    provide: NGX_UPLOAD_ROOT_GUARD,
                    useFactory: createNgxUploadRootGuard,
                    deps: [[NGX_LOGGER_OPTIONS, new Optional(), new SkipSelf()]]
                }
            ]
        };
    }
    ;
}
NgxUploadModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    ...ngxDeclarations
                ],
                exports: [
                    ...ngxDeclarations
                ],
                imports: [],
                entryComponents: [InputfileComponent]
            },] }
];
