// @wkoza/ngx-upload/esm2015/components/inputfile.component.js
import { Component, Injector, Optional, Renderer2, ViewChild } from '@angular/core';
import { FormGroupDirective, NgForm } from '@angular/forms';
import { HttpClientUploadService } from '../services/httpClientUpload.service';
export class InputfileComponent {
    constructor(injector, uploader, renderer, ngForm, formGroupDirective) {
        this.injector = injector;
        this.uploader = uploader;
        this.renderer = renderer;
        this.ngForm = ngForm;
        this.formGroupDirective = formGroupDirective;
        this.files = new Set();
        if (this.ngForm) {
            this.formGroup = ngForm.form;
        }
        else if (this.formGroupDirective) {
            this.formGroup = formGroupDirective.form;
        }
        else {
            this.formGroup = null;
        }
    }
    onFilesAdded() {
        this.uploader.addToQueue(this.file.nativeElement.files, this.formGroup, this.options);
        // Clear the previous input value
        this.file.nativeElement.value = '';
    }
    ngAfterViewInit() {
        if (this.options.multiple !== false)
            this.renderer.setProperty(this.file.nativeElement, 'multiple', 'multiple');
        if (this.options.accept)
            this.renderer.setProperty(this.file.nativeElement, 'accept', this.options.accept.join());
        if (this.options.capture)
            this.renderer.setProperty(this.file.nativeElement, 'capture', this.options.capture);
    }
}
InputfileComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-upload-inputfile',
                template: `
      <label class="input-file">
          <input type="file" #file (change)="onFilesAdded()">
          <ng-content></ng-content>
      </label>`,
                styles: ['input[type="file"] { display: none; } .input-file { width: 100%; }']
            },] }
];
InputfileComponent.ctorParameters = () => [
    { type: Injector },
    { type: HttpClientUploadService },
    { type: Renderer2 },
    { type: NgForm, decorators: [{ type: Optional }] },
    { type: FormGroupDirective, decorators: [{ type: Optional }] }
];
InputfileComponent.propDecorators = {
    file: [{ type: ViewChild, args: ['file', { static: true },] }]
};
