// @wkoza/ngx-upload/esm2015/directives/inputfile.directive.js
import { ComponentFactoryResolver, Directive, Injector, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { InputfileComponent } from '../components/inputfile.component';
import { ngxInputFileOptions } from '../utils/configuration.model';
export class NgxInputFileDirective {
    constructor(resolver, injector, vcRef, templateRef) {
        this.resolver = resolver;
        this.injector = injector;
        this.vcRef = vcRef;
        this.templateRef = templateRef;
    }
    ngOnInit() {
        const _contentViewRef = this.templateRef.createEmbeddedView(null);
        const factory = this.resolver.resolveComponentFactory(InputfileComponent);
        const component = this.vcRef.createComponent(factory, 0, this.injector, [_contentViewRef.rootNodes]);
        component.instance.options = (this.ngxInputFile) ? this.ngxInputFile : ngxInputFileOptions;
        _contentViewRef.detectChanges();
    }
}
NgxInputFileDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngxInputFile]'
            },] }
];
NgxInputFileDirective.ctorParameters = () => [
    { type: ComponentFactoryResolver },
    { type: Injector },
    { type: ViewContainerRef },
    { type: TemplateRef }
];
NgxInputFileDirective.propDecorators = {
    ngxInputFile: [{ type: Input }]
};
